### DISCUSSION TITLE
Option not to auto start CDN version

### DISCUSSION THREAD

**Discussion by mikestempien** [alpinejs/alpine#1776] [3. Feature ideas]
Since in the setup I'm working with I'm bundling all the libs into `vendor.js` and then init the scripts in `app.js`, I'm having a problem with Alpine auto-starting before I set up stores and components in `app.js`. Maybe we could check for some var ex: `window.someVarSoAlpineWontAutoStart = true` and don't start it automatically then?

See also: 
- https://github.com/alpinejs/alpine/discussions/1705
- https://github.com/alpinejs/alpine/discussions/1710

**Comment by danddanddand** [comment]
There was a PR to support this but was not added. You can read comments here https://github.com/alpinejs/alpine/pull/1721. 

I'd suggest you move all your stores and components into event listener using `document.addEventListener('alpine:init', () => { // yourStoresAndComponentsSetup() } )` and ensure that the vendor.js script has defer on it.

**Reply by mikestempien** [reply]
That's not possible I'm afraid - some of the scripts in app.js require some libs from vender.js to be already loaded.

**Comment by calebporzio** [comment]
@mikestempien - the file included from "import Alpine from alpinejs" shouldn't auto start - is that what you're experiencing?

**Reply by mikestempien** [reply]
Sorry, should have made it more clear that I'm referring to the CDN version.

**Reply by danddanddand** [reply]
So your vendor.js is a bundle of of CDN version of Alpine and other javascript assets? Can you just load alpine in its own script tag with defer on it?

What framework/tooling are you using?

**Reply by mikestempien** [reply]
Yep, and these are the limitations of the environment I'm often working with. Alpine 2.x works just fine and I guess I'll keep using it for the time being.

**Comment by calebporzio** [comment]
@mikestempien , you can import and use the module build to load Alpine from a script tag and not auto-initialize:
```js
<script type="module">
    import Alpine from 'https://unpkg.com/alpinejs@3.x.x/dist/module.esm.js'

    window.Alpine = Alpine

    Alpine.start()
</script>
```

If you need more control over starting Alpine, just leave out the `Alpine.start()` and call it somewhere else (from your bundle)

Hope this helps!
